components/ListItem.js

```diff
@@ -0,0 +1,16 @@
+import styled from "styled-components";
+
+export default function ListItem({ name, isChecked }) {
+  return (
+    <li>
+      <StyledLabel isChecked={isChecked}>
+        <input type="checkbox" defaultChecked={isChecked} />
+        {name}
+      </StyledLabel>
+    </li>
```

Comment:
You might use aria-label here instead of the StyledLabel, this nested writing is not common anymore. By this, you could get rid of one level of hierarchy and style the 'li' instead.